### DISCUSSION TITLE
Environment: In M1 MacOS 13.3.1, Rider, unable to start the project, the error is as follows:

### DISCUSSION THREAD

**Discussion by ILAOHE** [serenity-is/Serenity#6763] [Help]
info: Serenity.Web.NodeScriptRunner[0]
      > tsbuild:watch
      
info: Serenity.Web.NodeScriptRunner[0]
      > node ./tsbuild.js --watch
      
fail: Serenity.Web.NodeScriptRunner[0]
      /Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/node_modules/esbuild/lib/main.js:1833
      
fail: Serenity.Web.NodeScriptRunner[0]
                throw new Error(`
      
fail: Serenity.Web.NodeScriptRunner[0]
                      ^
      
fail: Serenity.Web.NodeScriptRunner[0]
      Error: 
      
fail: Serenity.Web.NodeScriptRunner[0]
      You installed esbuild for another platform than the one you're currently using.
      
fail: Serenity.Web.NodeScriptRunner[0]
      This won't work because esbuild is written with native code and needs to
      
fail: Serenity.Web.NodeScriptRunner[0]
      install a platform-specific binary executable.
      
fail: Serenity.Web.NodeScriptRunner[0]
      Specifically the "@esbuild/darwin-arm64" package is present but this platform
      
fail: Serenity.Web.NodeScriptRunner[0]
      needs the "@esbuild/darwin-x64" package instead. People often get into this
      
fail: Serenity.Web.NodeScriptRunner[0]
      situation by installing esbuild with npm running inside of Rosetta 2 and then
      
fail: Serenity.Web.NodeScriptRunner[0]
      trying to use it with node running outside of Rosetta 2, or vice versa (Rosetta
      
fail: Serenity.Web.NodeScriptRunner[0]
      2 is Apple's on-the-fly x86_64-to-arm64 translation service).
      
fail: Serenity.Web.NodeScriptRunner[0]
      If you are installing with npm, you can try ensuring that both npm and node are
      
fail: Serenity.Web.NodeScriptRunner[0]
      not running under Rosetta 2 and then reinstalling esbuild. This likely involves
      
fail: Serenity.Web.NodeScriptRunner[0]
      changing how you installed npm and/or node. For example, installing node with
      
fail: Serenity.Web.NodeScriptRunner[0]
      the universal installer here should work: https://nodejs.org/en/download/. Or
      
fail: Serenity.Web.NodeScriptRunner[0]
      you could consider using yarn instead of npm which has built-in support for
      
fail: Serenity.Web.NodeScriptRunner[0]
      installing a package on multiple platforms simultaneously.
      
fail: Serenity.Web.NodeScriptRunner[0]
      If you are installing with yarn, you can try listing both "arm64" and "x64"
      
fail: Serenity.Web.NodeScriptRunner[0]
      in your ".yarnrc.yml" file using the "supportedArchitectures" feature:
      
fail: Serenity.Web.NodeScriptRunner[0]
      https://yarnpkg.com/configuration/yarnrc/#supportedArchitectures
      
fail: Serenity.Web.NodeScriptRunner[0]
      Keep in mind that this means multiple copies of esbuild will be present.
      
fail: Serenity.Web.NodeScriptRunner[0]
      Another alternative is to use the "esbuild-wasm" package instead, which works
      
fail: Serenity.Web.NodeScriptRunner[0]
      the same way on all platforms. But it comes with a heavy performance cost and
      
fail: Serenity.Web.NodeScriptRunner[0]
      can sometimes be 10x slower than the "esbuild" package, so you may also not
      
fail: Serenity.Web.NodeScriptRunner[0]
      want to do that.
      
fail: Serenity.Web.NodeScriptRunner[0]
          at generateBinPath (/Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/node_modules/esbuild/lib/main.js:1833:17)
      
fail: Serenity.Web.NodeScriptRunner[0]
          at esbuildCommandAndArgs (/Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/node_modules/esbuild/lib/main.js:1912:33)
      
fail: Serenity.Web.NodeScriptRunner[0]
          at ensureServiceIsRunning (/Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/node_modules/esbuild/lib/main.js:2077:25)
      
fail: Serenity.Web.NodeScriptRunner[0]
          at Object.build (/Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/node_modules/esbuild/lib/main.js:1968:26)
      
fail: Serenity.Web.NodeScriptRunner[0]
          at build (file:///Users/heqinghuan/2023/node_modules/@serenity-is/tsbuild/dist/index.js:138:19)
      
fail: Serenity.Web.NodeScriptRunner[0]
          at file:///Users/heqinghuan/2023/tsbuild.js:3:7
      
fail: Serenity.Web.NodeScriptRunner[0]
          at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
      
fail: Serenity.Web.NodeScriptRunner[0]
      Node.js v20.0.0

**Comment by VictorTomaili** [comment]
```
You installed esbuild for another platform than the one you're currently using.
This won't work because esbuild is written with native code and needs to
install a platform-specific binary executable.
```
delete node_modules folder and Re install npm packages

**Reply by ILAOHE** [reply]
This solution has been tried many times on my Mac and is currently working fine on my windows computer. I have repeatedly deleted and executed 'npm run installl' but the problem has not been solved. I do not know if it is the Mac problem of M1 chip
